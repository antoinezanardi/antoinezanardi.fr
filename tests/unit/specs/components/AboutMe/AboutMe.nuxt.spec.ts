import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { AboutMe } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("About Me Component", () => {
  let wrapper: ReturnType<typeof mount<typeof AboutMe>>;

  async function mountAboutMeComponent(options: ComponentMountingOptions<typeof AboutMe> = {}): Promise<ReturnType<typeof mount<typeof AboutMe>>> {
    return mountSuspendedComponent(AboutMe, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountAboutMeComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});