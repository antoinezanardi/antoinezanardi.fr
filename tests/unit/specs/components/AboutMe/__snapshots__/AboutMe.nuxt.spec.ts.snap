// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Me Component > should match snapshot when rendered. 1`] = `
"<div id="about" class="section">
  <div class="container">
    <div class="card" data-aos="fade-up" data-aos-offset="10">
      <div class="row">
        <about-my-profile-stub class="col-lg-6 col-md-12"></about-my-profile-stub>
        <about-my-personal-info-stub class="col-lg-6 col-md-12"></about-my-personal-info-stub>
      </div>
    </div>
  </div>
</div>"
`;
